### ISSUE TITLE
TypeScript typings and possibly also Flow typings

### ISSUE THREAD

**Issue by jgonggrijp** [documentcloud/underscore-contrib#222]
It would be nice to have type declarations included within the repository and published to NPM, so that users don't need to rely on third-party type declarations (at least not for Contrib itself). This has been on @joshuacc's wish list but there was no ticket for it yet, at least not as far as I could tell.

I suggest to consider this a low priority enhancement. For maintenance reasons, it is probably best to wait with this at least until after #220 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45201.

A note to whoever might take up on this in the future: we have a convention in Underscore to write the source code proper in pure ES3 (with ES6 `import`/`export` for module linking only), so that it is ready to ship to all supported environments without any transpilation. I'd prefer to uphold this same convention in Contrib. So if you decide to add type declarations, please keep them separate from the source code proper.

(I don't mean to rule out that I might add the typings myself, by the way, but everyone is welcome to do it.)
